src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/StackItem.cs

```diff
@@ -3,68 +3,137 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Runtime.InteropServices;
 using Internal.IL;
 using Internal.TypeSystem;
 
 namespace Internal.Runtime.Interpreter
 {
-    internal abstract class StackItem
+    [StructLayout(LayoutKind.Explicit)]
+    internal unsafe struct StackItem
     {
-        public StackValueKind Kind { get; protected set; }
-        public WellKnownType Type { get; protected set; }
-    }
+        [FieldOffset(0)]
+        private StackValueKind _kind;
 
-    internal class StackItem<T> : StackItem
-    {
-        public T Value { get; }
+        [FieldOffset(8)]
+        private WellKnownType _type;
```

Comment:
Do we really need `WellKnownType` here? `StackValueKind` should be enough to capture the type information.